### PR TITLE
Unify minting process experience

### LINKED ISSUE #3703: Unify minting process experience - BalanceInput.vue
After we added additional checks in #3546, it would be nice if somebody could take a look at the overall way in which we handle minting process, more specifically, how do we check for things such as:

- [ ] insufficient balance
- [ ] invalid characters
- [ ] form validation in general

As an example, I've noticed one difference: On rmrk, insufficient balance means disabled submit button, on bsx insufficient balance means form validation error. iirc, @roiLeo was fan of the second solution, so we could move with that.

Point is to have similar UI/UX, it shouldn't change based on the chain (other than chain specific features)

// original comment: 

ok let's merge this since the bsx part seem so to be working (and that's why the issue was created), however ⚠️ we'll need more refactor, currently, the way we handle form validation / minting at rmrk doesn't match bsx at all (button to submit is disabled there in case of insufficient balance).

_Originally posted by @petersopko in https://github.com/kodadot/nft-gallery/issues/3546#issuecomment-1209075113_

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

### What's new?

- [x] PR closes #3703
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

RMRK mint:  
- [x] classic mint
- [x] simple mint
- [x] creative mint

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### Optional

- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases
- [ ] I've written some unit tests 🧪

### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### CODE REVIEW

**Comment by DeluneClaire** [comment]
Hi, I don't see anything as is on your screenshot.
Snímek obrazovky 2022-08-11 v 14 47 45

**Comment by DeluneClaire** [comment]
Still don't see the tab as yours...
Snímek obrazovky 2022-08-12 v 11 24 50

**Comment by petersopko** [comment]
> Still don't see the tab as yours... 

Create -> Simple
https://deploy-preview-3711--koda-nuxt.netlify.app/rmrk/mint

image

**Comment by DeluneClaire** [comment]
I have tried to mint the NFT, and I think this is a little bit strange. Since after some reloading of the page, you can see the NFT minted - which means it worked, but before you see the NFT it shows some error. 

https://user-images.githubusercontent.com/104791453/184615157-4283fcee-f786-4338-b734-eaee35db3f77.mov

**Comment by DeluneClaire** [comment]
So I would remove the "error" sign because it could confuse new users. And the NFT you can see only after reloading the page.

**Review by DeluneClaire — APPROVED** [review]
But otherwise I wrote, works for me:)

File: components/rmrk/Create/CreativeMint.vue [hunk 1]
```diff
@@ -2,62 +2,61 @@
...
+      key="edition"
+      class="mt-5"
+      :min="1"
+      :label="$t('mint.nft.edition.label')"
+      :message="$t('mint.nft.edition.message')"
+      :placeholder="$t('mint.nft.edition.placeholder')"
+      expanded />
+
+    <!-- <BalanceInput :step="0.1" @input="updateMeta" label="Price" expanded />
+>>>>>>> main
```
Comment (roiLeo) [inline components/rmrk/Create/CreativeMint.vue]:
```suggestion
>>>>>>> main
```

something went wrong 🍖

**Comment by Jarsen136** [comment]
> So I would remove the "error" sign because it could confuse new users.

Yes, but the warning maybe not be introduced by this PR IMO : )

**Comment by petersopko** [comment]
@Jarsen136 can you check out this one?

trying to mint NFT to existing collection on BSX without uploading file:
image

also, I've noticed that there are currently no checks once you upload the file? 
image

oh wow ok, apparently we can mint NFTs without name, is that ok @vikiival ?
https://deploy-preview-3711--koda-nuxt.netlify.app/bsx/u/bXkQe5T8pdkGn78sS6viEUs18wf1kAubDTQX1WWaEdMqDNyCN

**Comment by Jarsen136** [comment]
> @Jarsen136 can you check out this one?

There's something wrong with minting validation on BSX in the main branch.

Already fixed.
✅  minting collection on BSX
✅  minting token on BSX

**Review by petersopko — APPROVED** [review]
wfm

**Comment by yangwao** [comment]
pay 100

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $100 @ 54.61 USD/KSM ~ 1.831 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0x4f9bbfa29173a03933cfa0b693be576f59e0e8750478f6fb33d0050917f30119](https://kusama.subscan.io/extrinsic/0x4f9bbfa29173a03933cfa0b693be576f59e0e8750478f6fb33d0050917f30119)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**